src/Workspaces/Core/Portable/Workspace/Host/DocumentService/ISpanMappingService.cs

```diff
@@ -28,6 +28,11 @@ internal interface ISpanMappingService : IDocumentService
         /// </summary>
         bool SupportsMappingImportDirectives { get; }
 
+        Task<ImmutableArray<(string mappedFilePath, TextChange mappedTextChange)>> GetTextChangesAsync(
```

Comment:
Hmm, not a fan of using tuples for the array item type. Would it be possible to have a more concrete descriptive type?

Also, given the expectation in the method that mapping should occur that should probably exist somewhere in the name. Maybe `GetMappedTextChangesAsync`?

Reply:
Totally fair! I changed the tuple type to a record, hopefully that improves things.